src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +498,146 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
+        {
+            /*the DeleteDatabaseCommand command id should look like that: "db1/A_B_C/a6m8me9843jt"
+                ($"{databaseName}/{string.Join('_', nodes}")/{newGuid}
+            
+                for example:
+                    TestDB_1/A_C_B/sadjghsdkjfng
+                    TestDB_1/C/sadjghsdkjfng
+             */
+            nodes = null;
+            database = string.Empty;
+
+            var guidParts = guid.Split("/").ToArray();
+            if (guidParts.Length != 3)
+                return false;
+
+            nodes = guidParts[1].Split('_').Where(x => x != string.Empty).ToList();
+            database = guidParts[0];
+
+            return true;
+        }
+
+        public static bool TryGetLastRemoveNodeFromDatabaseCommand(ServerStore serverStore, 
+            string database, string removedNodeTag, long deleteIndex, out long index)
+        {
+            index = 0L;
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+                using var _ = Slice.From(context.Allocator, $"RemoveNodeFromDatabase/{database}/{removedNodeTag}/", out var guidPrfixSlice);
+
+                foreach (var (_, entryHolder) in table.SeekByPrimaryKeyPrefix(guidPrfixSlice, Slices.Empty, 0))
+                {
+                    var cmdIndex = ReadIndex(entryHolder);
+                    var guidFromCmd = ReadGuid(entryHolder);
+
+                    if (TryGetDeleteIndexFromDatabaseRemoveCommand(guidFromCmd, out var deleteIndexFromCmd) &&
+                        deleteIndex <= deleteIndexFromCmd && (deleteIndexFromCmd < index || index == 0)) // try to get the remove command with deleteIndexFromCmd which is the closest to 'deleteIndex' from above
+                    {
+                        index = cmdIndex;
+                        if (index == deleteIndex)
+                            break;
+                    }
+
+                }
+            }
+
+            return index != 0L;
+        }
+
+        private static bool TryGetDeleteIndexFromDatabaseRemoveCommand(string guid, out long deleteIndex)
+        {
+            /*the RemoveNodeFromDatabase command id should look like that: "RemoveNodeFromDatabase/A/17"
+                ($"RemoveNodeFromDatabase/{databaseName}/{removedNodeTag}/{deleteCmdIndex}"
+             */
+            deleteIndex = 0;
+
+            var guidParts = guid.Split('/');
+            if (guidParts.Length != 4 || guidParts[0] != "RemoveNodeFromDatabase")
+            {
+                return false;
+            }
+
+            deleteIndex = long.Parse(guidParts[3]);
+
+            return true;
+        }
+
+        public static bool TryGetLastRemoveNodeFromDatabaseCommand(ServerStore serverStore, string database, out long index)
```

Comment:
remove the `static` & the `serverStore`

Reply:
In RachisLogHistory every other method you pass a context, so I did the same here.. (I passed a context instead of serverStore)..